src/weights.jl

```diff
@@ -35,7 +36,33 @@ end
     W(v, sum(v))
 end
 
-Base.getindex(wv::W, ::Colon) where {W <: AbstractWeights} = W(copy(wv.values), sum(wv))
+Base.getindex(wv::AbstractWeights, ::Colon) = copy(wv)
+
+Base.copy(wv::W) where {W <: AbstractWeights} =
+    weightstype(W)(copy(wv.values), sum(wv))
+
+@propagate_inbounds function Base.view(wv::W, inds...) where {S <: Real, W <: AbstractWeights{S}}
+    @boundscheck checkbounds(wv, inds...)
+    @inbounds v = invoke(view, Tuple{AbstractArray, Vararg{Any}}, wv, inds...)
```

Comment:
why it is better to make such a view rather than just `v = view(wv.values, inds...)`? I understand the reason is that later you check if in the new object `values` are view of `AbstractWeights` to know you must recompute `sum`?

Reply:
This is to avoid the problem we discussed on Slack: if we merely make a view of `wv.values`, mutating the view will corrupt the parent as the sum will be out of sync with its contents.

(It's not needed to detect whether we must recompute the sum, as we could do that when `wv.values isa SubArray`, whatever the parent type.)

Reply:
Ah - right!